src/main.vala

```diff
@@ -77,26 +77,6 @@ class Vls.Server : Object {
     [CCode (has_target = false)]
     delegate void CallHandler (Vls.Server self, Jsonrpc.Server server, Jsonrpc.Client client, string method, Variant id, Variant @params);
 
-    private void log_handler (string? log_domain, LogLevelFlags log_levels, string message) {
-        string level = "";
-
-        if ((log_levels & LogLevelFlags.LEVEL_MASK) == LogLevelFlags.LEVEL_MASK)
-            level = "-ALL";
-        else if ((log_levels & LogLevelFlags.LEVEL_CRITICAL) != 0)
-            level = "-CRITICAL";
-        else if ((log_levels & LogLevelFlags.LEVEL_DEBUG) != 0)
-            level = "-DEBUG";
-        else if ((log_levels & LogLevelFlags.LEVEL_ERROR) != 0)
-            level = "-ERROR";
-        else if ((log_levels & LogLevelFlags.LEVEL_INFO) != 0)
-            level = "-INFO";
-        else if ((log_levels & LogLevelFlags.LEVEL_MESSAGE) != 0)
-            level = "-MESSAGE";
-        else if ((log_levels & LogLevelFlags.LEVEL_WARNING) != 0)
-            level = "-WARNING";
-        printerr ("%s: %s\n", log_domain == null ? @"vls$level" : log_domain, message);
```

Comment:
Hmm... the original intent of this function was to setup for changes that allow logging to a separate file. However, the `printerr()` portion in this handler may also capture any messages that are printed to stdout before the code on line 123 is executed, which would avoid corrupting the pipe.

Have you tested this and it works? I suspect you would have to do something like move the code starting at https://github.com/benwaffle/vala-language-server/blob/master/src/main.vala#L125 back to the top of `Vls.Server::Server()`

Reply:
I tested it. But it just do the same as original code so right now do not offer any improvement.

Just preparing to better filter info and low level messages (debug)

Reply:
It does make messages less verbose unless you have `G_MESSAGES_DEBUG=all`.